prime-router/build.gradle.kts

```diff
@@ -53,6 +53,11 @@ val azureFunctionsDir = "azure-functions"
 val primeMainClass = "gov.cdc.prime.router.cli.MainKt"
 val defaultDuplicateStrategy = DuplicatesStrategy.WARN
 azurefunctions.appName = azureAppName
+val appJvmTarget = "17"
+val javaVersion = when (appJvmTarget) {
```

Comment:
What was the reason for this `when` block?

Reply:
I based this primarily on Mo's work in #7539 where there was a similar structure. Primarily I'd like `appJvmTarget` to be the only value that has to be changed to target a specific Java version. To that end I modified the `when` block to include some potential future target versions.